src/main/java/uk/gov/companieshouse/api/accounts/controller/IntangibleAssetsController.java

```diff
@@ -0,0 +1,138 @@
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<IntangibleAssets> response = intangibleAssetsService.create(intangibleAssets, transaction,
+                    companyAccountId, request);
+            return apiResponseMapper.map(response.getStatus(), response.getData(), response.getErrors());
+
+        }
+        catch(DataException ex) {
+            LoggingHelper.logException(companyAccountId, transaction, "Failed to create intangible assets resource",
+                     ex, request);
+            return apiResponseMapper.getErrorResponse();
+        }
+
+    }
+
+    @GetMapping
+    public ResponseEntity get(@PathVariable("companyAccountId") String companyAccountId,
+                              HttpServletRequest request) {
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<IntangibleAssets> response = intangibleAssetsService.find(companyAccountId, request);
+            return  apiResponseMapper.mapGetResponse(response.getData(), request);
+        } catch (DataException ex) {
+            LoggingHelper.logException(companyAccountId, transaction, "Failed to retreive intangible assets resource", ex, request);
+            return apiResponseMapper.getErrorResponse();
+        }
+    }
+
+    @PutMapping
+    public ResponseEntity update(@RequestBody IntangibleAssets intangibleAssets, BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId, HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.INTANGIBLE_ASSETS_NOTE.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        if (bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<IntangibleAssets> response =
+                    intangibleAssetsService
+                        .update(intangibleAssets, transaction, companyAccountId, request);
+
+            return apiResponseMapper.map(response.getStatus(), response.getData(), response.getErrors());
+
+        } catch (DataException ex) {
+            LoggingHelper.logException(companyAccountId, transaction,
+                    "Failed to update tangible assets resource", ex, request);
```

Comment:
This error log reads 'tangible', not 'intangible'

Reply:
Well spotted B man